Storage/src/Connection/RetryTrait.php

```diff
@@ -173,13 +175,21 @@ private function retryDeciderFunction(
         $isIdempotent,
         $preconditionNeeded,
         $preconditionSupplied,
-        $maxRetries
+        $maxRetries,
+        $retryStrategy
     ) {
         // No retry if maxRetries reached
         if ($maxRetries <= $currentAttempt) {
             return false;
         }
 
+        switch ($retryStrategy) {
+            case Rest::ALWAYS_RETRY:
```

Comment:
I know we have discussed this offline but perhaps we should add a default of `RETRY_IDEMPOTENT`. Not only does it add visibility to the users, it makes it future proof(perhaps we change defaults or add options in the future).

Reply:
Okay.
It feels more logical now that I implemented the changes.